src/net_processing.cpp

```diff
@@ -2131,6 +2131,7 @@ static void ProcessGetCFHeaders(CNode& peer, CDataStream& vRecv, const CChainPar
         if (!filter_index->LookupFilterHeader(prev_block, prev_header)) {
             LogPrint(BCLog::NET, "Failed to find block filter header in index: filter_type=%s, block_hash=%s\n",
                          BlockFilterTypeName(filter_type), prev_block->GetBlockHash().ToString());
+            peer.fDisconnect = true;
```

Comment:
in commit fe2e6bb06705f72346ed636becc60c45075a451e

It looks like this could disconnect a permissioned peer, which is probably unwanted. See the second part of https://github.com/bitcoin/bitcoin/pull/19070#issuecomment-643474746

I'd prefer to wait until we can reply to the peer with a proper result.

Reply:
Reverted